api/routes/generate.js

```diff
@@ -10,35 +10,30 @@ var thunkify = require('thunkify');
 // CRX packaging module, instantiated with the `new` keyword
 var Extension = require('crx');
 
+var knownHosts = {
```

Comment:
The separation of logic for handling these custom cases is not ideal in my opinion -- it's much harder to maintain in comparison to how things work now, with the switch-case statement, where the hostname, title and icon overrides are in the same place together.